src/Compilers/CSharp/Test/Semantic/Semantics/PatternSubsumptionTests.cs

```diff
@@ -152,19 +149,19 @@ public static void Main()
         {
             case bool n:
                 break;
-            case ""foo"": // error: subsumed by previous case
-                break;
+            case ""foo"": // wrong type
+                break; // unreachable
         }
     }
 }";
             var compilation = CreateCompilationWithMscorlib45(source, options: TestOptions.DebugExe, parseOptions: patternParseOptions);
             Assert.True(compilation.GetDiagnostics().HasAnyErrors());
             compilation.VerifyDiagnostics(
                 // (10,18): error CS0029: Cannot implicitly convert type 'string' to 'bool'
-                //             case "foo": // error: subsumed by previous case
+                //             case "foo": // wrong type
                 Diagnostic(ErrorCode.ERR_NoImplicitConv, @"""foo""").WithArguments("string", "bool").WithLocation(10, 18),
                 // (11,17): warning CS0162: Unreachable code detected
-                //                 break;
+                //                 break; // unreachable
                 Diagnostic(ErrorCode.WRN_UnreachableCode, "break").WithLocation(11, 17)
```

Comment:
Are we reporting unreachable because the `case` expression was invalid or because all possible cases were handled earlier? We don't seem to report unreachable for non-pattern switch: `switch (true) { case false: case true: break; case "false": break; }`.